iam/src/providers/githubOrg.ts

```diff
@@ -0,0 +1,114 @@
+// ----- Types
+import type { RequestPayload, VerifiedPayload } from "@gitcoin/passport-types";
+import type { Provider, ProviderOptions } from "../types";
+import { requestAccessToken } from "./github";
+import axios from "axios";
+
+export type GithubTokenResponse = {
+  access_token: string;
+};
+
+export type GithubFindMyUserResponse = {
+  id?: string;
+  login?: string;
+  type?: string;
+};
+
+type GHUserRequestPayload = RequestPayload & {
+  org?: string;
+};
+
+type GithubMyOrg = {
+  providedOrg: string;
+  matchingOrg: string;
+};
+
+// Export a Github Provider to carry out OAuth and return a record object
+export class GithubOrgProvider implements Provider {
+  // Give the provider a type so that we can select it with a payload
+  type = "GithubOrg";
+
+  // Options can be set here and/or via the constructor
+  _options = {};
+
+  // construct the provider instance with supplied options
+  constructor(options: ProviderOptions = {}) {
+    this._options = { ...this._options, ...options };
+  }
+
+  // verify that the Github user is a memeber of the provided organization
+  async verify(payload: GHUserRequestPayload): Promise<VerifiedPayload> {
+    let valid = false,
+      validPayload: GithubMyOrg;
+
+    try {
+      validPayload = await verifyGithub(payload.proofs.code, payload.org);
+    } catch (e) {
+      return { valid: false };
+    } finally {
+      valid = validPayload && validPayload.matchingOrg === validPayload.providedOrg;
+    }
+
+    return {
+      valid: valid,
+      record: {
+        org: validPayload.matchingOrg,
```

Comment:
To make the PII hash more useful it might be an idea to drop the users handle here too. With this as is, if multiple users are attesting to the same org they will have the same hash so we won't be able to spot duplicates

Reply:
Great callout! I just included that